3rdparty/openssl/CMakeLists.txt

```diff
@@ -891,8 +983,10 @@ add_enclave_library(
   openssl/crypto/x509v3/v3_sxnet.c
   openssl/crypto/x509v3/v3_tlsf.c
   openssl/crypto/x509v3/v3_utl.c
-  openssl/crypto/x509v3/v3err.c
-  ${PLATFORM_SRC})
+  openssl/crypto/x509v3/v3err.c)
+
+add_enclave_library(opensslcrypto_asm OBJECT ${GENERATED_ASM_SRC}
```

Comment:
Is there are reason for creating a separate library for this that needs to be published instead of just linking the obj files directly into the existing opensslcrypto lib?

Reply:
I simply want to call out that these assembly files   are not the same as what come from OpenSSL (either generated with the modified peel script or copied from sgxssl repo).

Reply:
I guess I'm not sure I understand how putting it into a separate library helps? It's a non-optional library if you're using OpenSSL in enclaves so it always needs to be included anyway, and it creates an additional expansion of the matrix from 2 to four new libraries for OpenSSL (2 new LVI variant libraries) 

Also one additional clarification: the 3 manually edited .s files from Intel don't actually contain the LVI mitigation, correct? They just make it possible for the mitigating assembler to identify the location that needs to be patched by removing the use of constant encodings, correct?

Reply:
Revert the changes of adding a new library as you suggest.

Regarding your question, it's partially correct. These 3 files contain lfence for indirect branch instructions (which are still constant encoded) but not for ret instructions (handled by the modified perl script). But for the simplicity of the build flow, I use them for both normal and LVI mitigated libraries.